src/components/App.js

```diff
@@ -141,6 +154,40 @@ class App extends Component<Props, State> {
     shortcuts.off(L10N.getStr("gotoLineModal.key2"), this.toggleQuickOpenModal);
 
     shortcuts.off("Escape", this.onEscape);
+
+    shortcuts.off("CmdOrCtrl+Left", this.goToPreviousTab);
+    shortcuts.off("CmdOrCtrl+Right", this.goToNextTab);
+  }
+
+  goToNextOrPreviousTab(e, goNext) {
+    e.preventDefault();
+
+    const { selectedSource, tabSources, selectSpecificSource } = this.props;
+
+    // There needs to be multiple sources for any action to be taken
+    if (!selectedSource || this.props.tabSources.size < 2) {
+      return;
+    }
+
+    // Get the index of the source to be selected
+    const currentIndex = tabSources.indexOf(this.props.selectedSource);
+    if (currentIndex < 0) {
+      return;
+    }
+
+    // Calculate the next index
+    let nextIndex = 0;
+
+    // Next
+    if (goNext) {
+      nextIndex = currentIndex === tabSources.size - 1 ? 0 : currentIndex + 1;
+    } else {
+      // Previous
+      nextIndex = currentIndex === 0 ? tabSources.size - 1 : currentIndex - 1;
+    }
```

Comment:
I've found over time that this logic is better in an action:

1. the component logic is easier to understand (get the event and surrounding state and pass it to the action)
2. the action is easier to understand because it is not wrapped in a class and its clear how it will be fired
3. the action is easier to test independently
4. the action has access to `getState` which keeps the component from needing more props via connect